engine/src/main/java/io/zeebe/engine/processing/streamprocessor/sideeffect/SideEffects.java

```diff
@@ -7,9 +7,16 @@
  */
 package io.zeebe.engine.processing.streamprocessor.sideeffect;
 
+import java.util.function.Consumer;
+
 /** A chain of side effects that are executed/flushed together at the end of the processing. */
-public interface SideEffects {
+public interface SideEffects extends Consumer<SideEffectProducer> {
```

Comment:
It's not clear to me why SideEffects should extend Consumer. I see a reference to the `accept` method is passed along sometimes, but the `add` method could be used instead because it has the same signature. Perhaps I missed something, could you please explain?

Reply:
Sorry, that was a half baked attempt to unify `SideEffects` and `Consumer<SideEffectProvider`. These two interfaces in our code base represent the same concept. I made some half-baked attempts to unify them. But you are right, in this current form it is not adding anything, so I removed it again.